compiler/src/iree/compiler/Preprocessing/Passes.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_PREPROCESSING_PASSES_H_
+#define IREE_COMPILER_PREPROCESSING_PASSES_H_
+
+#include <functional>
+
+#include "iree/compiler/Pipelines/Options.h"
+#include "mlir/Pass/PassManager.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+
+/// Adds a set of passes to the given pass manager that run preprocessing
```

Comment:
Are these the first codegen/high-level codegen focussed passes? (Preprocessing just makes think of it even earlier in the pipeline, e.g., "preprocessing for X")

Reply:
These are passes to be run very early on in IREE. Effectively after conversion to Linalg but before anything else that is part of "core IREE" runs. So its "preprocessing for X" kind of thing for custom use cases.